mempool/clist_mempool.go

```diff
@@ -700,20 +772,18 @@ type recheck struct {
 	recheckFull   atomic.Bool     // whether rechecking TXs cannot be completed before a new block is decided
 }
 
-func newRecheck() *recheck {
-	return &recheck{
-		doneCh: make(chan struct{}, 1),
-	}
-}
-
 func (rc *recheck) init(first, last *clist.CElement) {
 	if !rc.done() {
 		panic("Having more than one rechecking process at a time is not possible.")
 	}
 	rc.cursor = first
 	rc.end = last
+	rc.doneCh = make(chan struct{})
 	rc.numPendingTxs.Store(0)
 	rc.isRechecking.Store(true)
+	rc.recheckFull.Store(false)
```

Comment:
Why is this added here?

Reply:
We are calling recheck.init on multiple instances. It's just to be sure that all variables are initialized in the correct state. We should also clean up the variables when a recheck instance finishes.